### PR TITLE
mgr/dashboard: Add i18n validation script

### PR DESCRIPTION
**Pull request by tspmelo**
This will check if there are new strings and ask the developer to commit those.

`Signed-off-by: Tiago Melo <tmelo@suse.com>`

### CODE REVIEW

File: src/pybind/mgr/dashboard/HACKING.rst [hunk 1]
```diff
@@ -249,6 +249,10 @@ parse the TypeScript files.
 When the command ran successfully, it should have created or updated the file
 ``src/locale/messages.xlf``.
 
+To make sure this file is always updated in master we have created a script,
+``run-frontend-i18n.sh``, that when executed will fail if it finds uncommited
```
Comment (callithea) [inline src/pybind/mgr/dashboard/HACKING.rst]:
```suggestion
``run-frontend-i18n.sh``, that when executed will fail if it finds uncommitted
```

File: src/pybind/mgr/dashboard/HACKING.rst [hunk 2]
```diff
@@ -249,6 +249,10 @@ parse the TypeScript files.
 When the command ran successfully, it should have created or updated the file
 ``src/locale/messages.xlf``.
 
+To make sure this file is always updated in master we have created a script,
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard/HACKING.rst]:
```suggestion
To make sure this file is always up to date in the master branch, we have created a script
```

File: src/pybind/mgr/dashboard/HACKING.rst [hunk 3]
```diff
@@ -249,6 +249,10 @@ parse the TypeScript files.
...
+``run-frontend-i18n.sh``, that when executed will fail if it finds uncommitted
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard/HACKING.rst]:
```suggestion
``run-frontend-i18n.sh`` that will fail if it finds uncommitted translations.
```

File: src/pybind/mgr/dashboard/run-frontend-i18n.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  . $CEPH_ROOT/build/src/pybind/mgr/dashboard/node-env/bin/activate ||
+  echo 'Info: missing node-env'
+fi
+
+npm run i18n
+
+count=`git status | grep "messages.xlf" -c`
+
+if [ $count == 1 ]; then
+  echo  "Please run 'npm run i18n' and commit the modified 'messages.xlf' file."
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard/run-frontend-i18n.sh]: At this point `npm run i18n` has already been run and `messages.xlf` is up to date, isn't it?
```suggestion
  echo  "The translation template file 'messages.xlf' has been modified. Please commit the changes."
```

File: src/pybind/mgr/dashboard/run-frontend-i18n.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+cd frontend
+
+if [ `uname` != "FreeBSD" ]; then
+  . $CEPH_ROOT/build/src/pybind/mgr/dashboard/node-env/bin/activate ||
+  echo 'Info: missing node-env'
+fi
+
+npm run i18n
+
+count=`git status | grep "messages.xlf" -c`
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard/run-frontend-i18n.sh]: This looks somewhat brittle. Wouldn't it be better to use git to check the modification status of this file directly, e.g. via `git status -s messages.xlf`

File: src/pybind/mgr/dashboard/HACKING.rst (hunk 2 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard/HACKING.rst]: Done.

File: src/pybind/mgr/dashboard/HACKING.rst (hunk 3 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard/HACKING.rst]: Done.

File: src/pybind/mgr/dashboard/run-frontend-i18n.sh (hunk 4 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard/run-frontend-i18n.sh]: This script is more focused in being used in jenkins, so this message is going to be read after a failing `ceph dashboard tests`. Then the developer will need to run the script locally.

**Comment by tspmelo** [comment]
I moved the validation to the existing `run-frontend-unittests.sh`, this will reduce the number of scripts we need to run.
The change is in a temporary commit, if everyone agrees with the changes I will squash them.

**Comment by tspmelo** [comment]
@p-na I added a new validation to `run-frontend-unittests.sh` that will check if the strings have any extra white-space in the beginning or end of the string.

**Comment by tchaikov** [comment]
retest this please

**Comment by LenzGr** [comment]
retest this please

**Pull request merged**